lib/xapi-stdext-date/date.ml

```diff
@@ -44,50 +45,42 @@ type iso8601 = Ptime.t
   match x |> Ptime.of_rfc3339 |> Ptime.rfc3339_error_to_msg with
   | Error (`Msg e) -> invalid_arg (Printf.sprintf "date.ml:of_string: %s" e)
   | Ok (t, tz, _)  -> match tz with
-                      | None | Some 0 -> t
+                      | None | Some 0 -> (t, UTC)
                       | Some _        -> invalid_arg (Printf.sprintf "date.ml:of_string: %s" x)
 
-let to_string t =
-  Ptime.to_rfc3339 ~tz_offset_s:0 (* to ensure Z printed, rather than +00:00 *) t |>
-  Astring.String.filter (fun char -> char <> '-') (* remove dashes for backwards compatibility *)
-
-let of_float x =
-  let time = Unix.gmtime x in
-  Printf.sprintf "%04d-%02d-%02dT%02d:%02d:%02dZ"
-    (time.Unix.tm_year+1900)
-    (time.Unix.tm_mon+1)
-    time.Unix.tm_mday
-    time.Unix.tm_hour
-    time.Unix.tm_min
-    time.Unix.tm_sec |> of_string
+let to_string (t, zone) =
+  let t_string =
+    (* to ensure Z printed, rather than +00:00 *)
+    Ptime.to_rfc3339 ~tz_offset_s:0 t
```

Comment:
For Local I think you need to get the current time zone offset so that it gets adjusted correctly.